### ISSUE TITLE
Package request: dropbear-hacks

### ISSUE THREAD

**Issue by lars18th** [termux/termux-packages#2241]
Hi,

Please, see this package: http://github.com/zcutlip/dropbear-hacks

What you get applying these hacks to dropbear?

- Server Master Password: You can use SSH with a password (clean pass running "dropbear-hacks -Y mypass"
- Root & Shell replacement: using FAKE_ROOT and ALT_SHELL environment variables.

I know that CERTIFICATES is the best solution for SSH access. However, if you incorporate these hacks to the current dropbear package, or add a new one called "dropbear-hacks", then we can use SSH with password in very special cases.

I hope you like to consider it.
Regards.

**Comment by dAtiuy** [comment]
Wrong place. This is issues not pull requests...

**Comment by lars18th** [comment]
> Wrong place. This is issues not pull requests...

Sorry! This is not a PR... This is a "new" package that I found it. And my suggestion is to incorporate this package (`dropbear-hacks`) or better, use it to improve the current `dropbear` for Termux. I'm not the developer of this package.

In any case. Why some people disaggree with this? Nothing changes... it only _adds_ options to the user.

**Comment by sdrausty** [comment]
```
$ pkg list-all|grep bear
.
.
.

dropbear/stable,now 2017.75-2 aarch64 [installed]
```
Nice; We have dropbear, dropbear-hacks looks like a nice complement.

> ... it only *adds* options to the user.

Ditto; the more, the merrier.

**Comment by lars18th** [comment]
Hi @sdrausty ,

I kown that the dropbear package is in the repository!
http://github.com/termux/termux-packages/tree/master/packages/dropbear

And I'm glad you found it interesting.
We have two options:

1.  Incorporate the dropbear-hacks inside the current patches for dropbear;
2.  Create a new package.

What you prefer?

**Comment by sdrausty** [comment]
Hello @lars18th,

Option two (2) sounds great!  Why?  Size, modularity, workflow and stabilty, how about your?  Two is better than one reasoning too.  Which option is your preference, and why?

**Comment by lars18th** [comment]
Hi,

Let me to explain some questions:

1) I'm not a developer (almost of termux). So, I'm only **suggesting** to _improve_ the dropbear package.

2) The `dropbear` package is already supported in Termux. And any Termux package is made using "patches" over original/generic packages. So, my suggestion is to _enhance_ the patches incorporating the modifications from `dropbear-hacks`. I prefer this over create a new package, as the changes are small (but useful).

3) At time Termux has two SSH servers: `sshd` (openSSH) and `dropbear`. The advantage of merge the `dropbear-hacks` mods is quite simple: you can run one SSH server with **PASSWORD** support. You not need at all to run it in this mode (the server _can be_ configured, but is not required). And this only applies to the dropbear binary, and not to the original sshd.

So, in summary: I feel interesting to incorporate (merge) the `dropbear-hacks` to the current Termux's `dropbear` package. I hope some developer will like to do it.

Regards.

**Comment by lars18th** [comment]
Hi,

No interest to add this `dropbear-hacks` to the official repository?

**Comment by eadmaster** [comment]
any update on this?
do you have a compiled package i can install manually and try?

**Comment by lars18th** [comment]
Hi @eadmaster ,

I'm not a developer of Termux. So, someone else needs to incorporate the patches from `dropbear-hacks` in the Termux's `dropbear` package or create a fork of it.

Regards.

**Comment by lars18th** [comment]
Hi @xeffyr ,

Good job! Thank you. :+1: 

I close the issue. :wink:
